qiskit/circuit/instruction.py

```diff
@@ -136,34 +137,36 @@ def params(self, parameters):
         self._params = []
         for single_param in parameters:
             # example: u2(pi/2, sin(pi/4))
-            if isinstance(single_param, (ParameterExpression, sympy.Basic)):
+            if isinstance(single_param, (ParameterExpression)):
                 self._params.append(single_param)
             # example: OpenQASM parsed instruction
             elif isinstance(single_param, node.Node):
-                self._params.append(single_param.sym())
+                self._params.append(single_param.num())
             # example: u3(0.1, 0.2, 0.3)
             elif isinstance(single_param, (int, float)):
-                self._params.append(sympy.Number(single_param))
+                self._params.append(single_param)
             # example: Initialize([complex(0,1), complex(0,0)])
             elif isinstance(single_param, complex):
-                self._params.append(single_param.real +
-                                    single_param.imag * sympy.I)
+                self._params.append(single_param)
             # example: snapshot('label')
             elif isinstance(single_param, str):
-                self._params.append(sympy.Symbol(single_param))
+                self._params.append(float(single_param))
             # example: numpy.array([[1, 0], [0, 1]])
             elif isinstance(single_param, numpy.ndarray):
                 self._params.append(single_param)
-            # example: sympy.Matrix([[1, 0], [0, 1]])
-            elif isinstance(single_param, sympy.Matrix):
-                self._params.append(single_param)
-            elif isinstance(single_param, sympy.Expr):
-                self._params.append(single_param)
             elif isinstance(single_param, numpy.number):
-                self._params.append(sympy.Number(single_param.item()))
+                self._params.append(single_param)
```

Comment:
Are numpy scalars valid parameters (or should they be)? I'm wondering about cases where we have code like `isinstance(param, float)`.

Maybe still `single_param.item()` (though there are a couple of cases where it may not return a python scalar,  https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.item.html )?